plugins/offers.c

```diff
@@ -159,8 +164,38 @@ struct decodable {
 	struct tlv_invoice *invoice;
 	struct tlv_invoice_request *invreq;
 	struct rune *rune;
+	u8 *emergency_recover;
 };
 
+static u8 *encrypted_decode(const tal_t *ctx, const char *str, char **fail) {
+	if (strlen(str) < 8) {
+		*fail = tal_fmt(ctx, "invalid payload");
+		return NULL;
+	}
+
+	size_t hrp_maxlen = strlen(str) - 6;
+	char *hrp = tal_arr(ctx, char, hrp_maxlen);
+
+	size_t data_maxlen = strlen(str) - 8;
+	u5 *data = tal_arr(ctx, u5, data_maxlen);
+	size_t datalen = 0;
+
+	if (bech32_decode(hrp, data, &datalen, str, (size_t)-1)
+		== BECH32_ENCODING_NONE)
+		goto fail;
+
+	if (!streq(hrp, "clnemerg"))
+		goto fail;
+
+	u8 *data8bit = tal_arr(data, u8, 0);
+	bech32_pull_bits(&data8bit, data, datalen*5);
+
+	return data8bit;
+fail:
+	tal_free(data);
+	return NULL;
+}
+
```

Comment:
You changed this to set `*fail`, but it doesn't in various paths?

Reply:
Thanks for pointing out.